homeassistant/components/plex/server.py

```diff
@@ -57,7 +61,18 @@ def __init__(self, hass, server_config, options=None):
         plexapi.myplex.BASE_HEADERS = plexapi.reset_base_headers()
         plexapi.server.BASE_HEADERS = plexapi.reset_base_headers()
 
-    def connect(self):
+    async def async_setup_websocket_callback(self):
```

Comment:
Maybe this should not be a method here inside the plex server but just a function in the component. We set up the websocket in the component. The plex server doesn't know about the websocket so it doesn't seem necessary to keep the callback setup here.

If we call it from `async_setup_entry` we can await it directly.

Reply:
That would be much cleaner. Thanks for the idea!